File: src/main/java/order/Order.java
package order;

import item.Item;
import lombok.*;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.List;

@EqualsAndHashCode
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Order {
    private long id;
    @EqualsAndHashCode.Exclude
    @NotNull
    @Size(min = 2)
    private String orderNumber;
    @EqualsAndHashCode.Exclude
    @Valid
    private List<Item> orderRows;

    public void addItem(Item item){
        if (item != null){
            orderRows.add(item);
        }
    }
}

File: src/main/kotlin/order/Order.kt
package order

import item.Item
import lombok.*
import javax.validation.Valid
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size


data class Order(
        var id: Long = 0,

        @field:EqualsAndHashCode.Exclude
        @field:NotNull
        @field:Size(min = 2)
        var orderNumber: String? = null,

        @field:EqualsAndHashCode.Exclude
        @field:Valid
        var orderRows:  MutableList<Item>? = null,
)
